joomla/joomla-cms
# Allow Joomla to be extended with JDEBUG defined by developers

Pull Request for Issue #15827 .

### Summary of Changes

Allow Joomla to be extended with JDEBUG defined by developers when in a sub folder without FOF moaning and fatal erroring. 

### Testing Instructions

Apply PR and check Joomla operates as normal.

Create api.php in Joomla tmp folder with content 


```php
<?php
define('_JEXEC', 1);
define('JPATH_BASE', realpath('../'));
ini_set('display_errors', '1');
error_reporting(E_ALL);
define('JDEBUG', 1);
include_once JPATH_BASE . '/includes/defines.php';
include_once JPATH_BASE . '/includes/framework.php';
include_once JPATH_BASE . '/libraries/joomla/factory.php';
```



### Expected result

No errors on running 

### Actual result

```
PHP Fatal error:  Uncaught Error: Call to a member function logAddLogger() on null in /Users/phil/Sites/joomla-cms/libraries/fof/include.php:26
Stack trace:
#0 /Users/phil/Sites/joomla-cms/libraries/cms.php(45): include_once()
#1 /Users/phil/Sites/joomla-cms/includes/framework.php(23): require_once('/Users/phil/Sit...')
#2 /Users/phil/Sites/joomla-cms/tmp/api.php(8): include_once('/Users/phil/Sit...')
#3 {main}
  thrown in /Users/phil/Sites/joomla-cms/libraries/fof/include.php on line 26
```

### Documentation Changes Required

None
